### DISCUSSION TITLE
Python code as command when using hook plugin?

### DISCUSSION THREAD

**Discussion by sandersantema** [beetbox/beets#3944] [Q&A]
Is it possible to use python code for the command to be triggered by a hook instead of a shell command just like you can for paths using the inline plugin? Given that I'll be writing some python code to execute when a hook is triggered it'd easier to directly use python code instead of parsing arguments.

**Comment by jackwilsdon** [comment]
At that point it sounds like your best bet is just to write your own beets plugin - take a look at the [writing plugins](https://beets.readthedocs.io/en/stable/dev/plugins.html) documentation on how to get started.
